realm/realm-library/src/androidTest/kotlin/io/realm/UnmanagedSetTester.kt

```diff
@@ -0,0 +1,296 @@
+        realmSet.addAll(values)
+        values.forEach { value ->
+            assertTrue(realmSet.contains(value))
+        }
+    }
+
+    override fun iterator() {
+        val realmSet = RealmSet<T>()
+        assertNotNull(realmSet.iterator())
+    }
+
+    override fun toArray() {
+        val realmSet = RealmSet<T>()
+        val emptyArray = realmSet.toArray()
+        assertEquals(0, emptyArray.size)
+
+        realmSet.addAll(values)
+        val fullArray = realmSet.toArray()
+        assertEquals(values.size, fullArray.size)
+    }
+
+    override fun add() {
+        val realmSet = RealmSet<T>()
+        assertTrue(realmSet.isEmpty())
+        values.forEach { value ->
+            realmSet.add(value)
+        }
+        assertFalse(realmSet.isEmpty())
+        assertEquals(values.size, realmSet.size)
+    }
+
+    override fun remove() {
+        val realmSet = RealmSet<T>()
+        realmSet.addAll(values)
+        assertEquals(values.size, realmSet.size)
+
+        assertTrue(realmSet.remove(values[0]))
+        assertEquals(values.size - 1, realmSet.size)
+        assertFalse(realmSet.remove(notPresentValue))
+    }
+
+    override fun containsAll() {
+        val realmSet = RealmSet<T>()
+        realmSet.addAll(values)
+        assertTrue(realmSet.containsAll(values))
+    }
+
+    override fun addAll() = Unit        // Tested multiple times in the other functions
+
+    override fun retainAll() {
+        val realmSet = RealmSet<T>()
+        realmSet.addAll(values)
+
+        // Test intersection with present value
+        val presentValues = values.subList(0, 1)
+        assertTrue(realmSet.retainAll(presentValues))
+        assertEquals(presentValues.size, realmSet.size)
+
+        // Test intersection with non-present value
+        val differentRealmSet = RealmSet<T>()
+        differentRealmSet.addAll(values)
+        assertTrue(differentRealmSet.retainAll(listOf(notPresentValue)))
+        assertTrue(differentRealmSet.isEmpty())
+    }
+
+    override fun removeAll() {
+        val realmSet = RealmSet<T>()
+        realmSet.addAll(values)
+        assertFalse(realmSet.isEmpty())
+        realmSet.removeAll(values)
+        assertTrue(realmSet.isEmpty())
+
+        val differentRealmSet = RealmSet<T>()
+        differentRealmSet.addAll(values)
+        assertFalse(differentRealmSet.isEmpty())
+        differentRealmSet.removeAll(listOf(notPresentValue))
+        assertEquals(values.size, differentRealmSet.size)
+    }
+
+    override fun clear() {
+        val realmSet = RealmSet<T>()
+        realmSet.addAll(values)
+        assertFalse(realmSet.isEmpty())
+        realmSet.clear()
+        assertTrue(realmSet.isEmpty())
+    }
+
+    override fun freeze() {
+        assertFailsWith<UnsupportedOperationException> {
+            RealmSet<T>().freeze()
+        }
+    }
+}
+
+fun unmanagedSetFactory(): List<SetTester> {
+    val primitiveTesters: List<SetTester> = SetSupportedType.values().mapNotNull { supportedType ->
```

Comment:
Yet again I would have done it by an _all-types_ global test enumeration instead with an `isSetSupported()` method and then verify `assertFalse(supportedType.isSetSupported` in the else clause.